pylitterbot/robot/litterrobot4.py

```diff
@@ -744,6 +752,40 @@ async def update_firmware(self) -> bool:
         firmware = data.get("data", {}).get("litterRobot4TriggerFirmwareUpdate", {})
         return bool(firmware.get("isUpdateTriggered", False))
 
+    async def toggle_hopper(self, is_removed: bool) -> bool:
+        """Enables/Disables the LitterHopper. A disabled hopper is synonymous with being removed. Returns `True` if request was successful."""
+        data = await self._post(
+            json={
+                "query": """
+                    mutation ToggleHopper($serial: String!, $isRemoved: Boolean!) {
+                        toggleHopper(serial: $serial, isRemoved: $isRemoved) {
+                            success
+                        }
+                    }
+                """,
+                "variables": {"serial": self.serial, "isRemoved": is_removed},
+            }
+        )
+        data = cast(dict, data)
+        toggle_hopper = data.get("data", {}).get("toggleHopper", {})
+        is_success = bool(toggle_hopper.get("success", False))
+        if is_success:
+            # data is now stale, hopper-related properties will have changed as
+            # a consequence. This mutation doesn't expose robot data for a
+            # partial update, so we refresh the data instead.
+            #
+            # Unfortunately `hopperStatus` doesn't instantaneously update to
+            # reflect the changes from the mutation. This is likely due to the
+            # LitterHopper device reporting back its own status back to the
+            # API. Wait 5s
+            if "PYTEST_CURRENT_TEST" not in os.environ:
+                _LOGGER.debug(
+                    f"sleeping {POST_MUTATION_REFRESH_DELAY_SECONDS}s before refreshing data"
+                )
+                await asyncio.sleep(POST_MUTATION_REFRESH_DELAY_SECONDS)
```

Comment:
We shouldn't force a 5s sleep to retrieve the status. This will cause many implementations to hang while waiting for a response when it is not needed. Instead, I would prefer to temporarily set the hopper status based on if we get a success response back.